misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/MockAnnotationProcessor.java

```diff
@@ -122,6 +123,21 @@ public void process(SimpleVariableNode simpleVariableNode, List<AnnotationAttach
         }
     }
 
+    @Override
+    public void process(FunctionNode functionNode, List<AnnotationAttachmentNode> annotations) {
+        // Remove the duplicated annotations.
+        annotations = annotations.stream().distinct().collect(Collectors.toList());
+        // Check for usage of mock annotation on a function which is deprecated.
+        for (AnnotationAttachmentNode attachmentNode : annotations) {
+            String annotationName = attachmentNode.getAnnotationName().getValue();
+            if (MOCK_ANNOTATION_NAME.equals(annotationName)) {
+                // Throw an error saying its not a MockFunction object
+                diagnosticLog.logDiagnostic(DiagnosticSeverity.ERROR, attachmentNode.getPosition(),
+                        "Annotation can only be attached to a test:MockFunction object");
```

Comment:
Shall we say "Mock annotation can only be attached to a test:MockFunction object"